File: Ex6/jmmakiik/Problem2/reading.py
import ROOT
from array import array

ffile = ROOT.TFile.Open("writing.root")
readtree = ffile.Get("MyTree")   

variable = array("f", [0])
branch = readtree.GetBranch("Random_Gaussian_numbers")
branch.SetAddress(variable)

histo = ROOT.TH1F("Histogram","",100,-5,5)
histo.SetStats(0)

for i in range(0,readtree.GetEntries(),1):
    readtree.GetEvent(i)
    histo.Fill(variable[0])
    
canvas = ROOT.TCanvas("reading","",500,500)
canvas.cd()
gaus = ROOT.TF1("gaus","gaus",-5,5)
gaus.SetLineWidth(4)
    
histo.SetLineColor(1)
histo.SetLineWidth(2)
histo.SetFillColor(5)
histo.SetTitle("Histogram of random numbers fitted with Gaussian")
histo.GetXaxis().SetTitle("Random number")
histo.GetYaxis().SetTitle("Number of occurences")
histo.Fit(gaus)
histo.Draw("")
canvas.Print("histogram.C")
canvas.Print("histogram.png")

File: Ex6/jmmakiik/Problem1/reading.cxx
#include "TTree.h"
#include "TBranch.h"
#include "TFile.h"
#include "TCanvas.h"
#include "TH1F.h"

using namespace std;

int main(){

    TFile f("writing.root");
    TTree* readtree = (TTree*)f.Get("MyTree");    

    Float_t variable[0];
    readtree->SetBranchAddress("Random_Gaussian_numbers",&variable);
//    readtree.Branch.SetAddress(&variable);
    
    TH1F histo("Histogram","",100,-5,5);
    histo.SetStats(0);
    
    Int_t nevents = readtree->GetEntries();

    for(Int_t i = 0; i < nevents; ++i){
      readtree->GetEntry(i);
      histo.Fill(variable[0]);
    }
    
    TCanvas canvas("reading","",500,500);
    canvas.cd();
    
    histo.SetLineColor(1);
    histo.SetLineWidth(2);
    histo.SetFillColor(5);
    histo.SetTitle("Histogram of random numbers fitted with Gaussian");
    histo.GetXaxis()->SetTitle("Random number");
    histo.GetYaxis()->SetTitle("Number of occurences");
    histo.Fit("gaus");
    histo.Draw("");
    canvas.Print("histogram.C");
    canvas.Print("histogram.png");
}

//I wasn't able to change the linewidth of the fit as with Python or last week with ROOT only. 
//It always gave me an error so decided not to include it on this exercise.
